src/main/org/codehaus/groovy/runtime/DefaultGroovyMethods.java

```diff
@@ -16045,6 +16048,51 @@ public void remove() {
     }
 
     /**
+     * Returns a <code>BufferedIterator</code> that allows examining the next element without
+     * consuming it.
+     * <pre class="groovyTestCase">
+     * assert [1, 2, 3, 4].iterator().buffered().with { [head(), toList()] } == [1, [1, 2, 3, 4]]
+     * </pre>
+     *
+     * @param self an iterator object
+     * @return a BufferedIterator wrapping self
+     * @since 2.5.0
+     */
+    public static <T> BufferedIterator<T> buffered(Iterator<T> self) {
+        return new IteratorBufferedIterator<T>(self);
```

Comment:
In case `self` is a `BufferedIterator` should this return `self` instead of re-wrapping?